runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/Work.java

```diff
@@ -88,7 +87,7 @@ private Work(
     this.watermarks = watermarks;
     this.clock = clock;
     this.startTime = clock.get();
-    this.totalDurationPerState = new EnumMap<>(LatencyAttribution.State.class);
+    this.totalDurationPerState = new HashMap<>();
```

Comment:
EnumMap is constructing a keyUniverse array in the constructor which shows up in profiles, replacing it with hashmap to speed up work submit loop. 
<img width="1693" alt="Screenshot 2025-01-22 at 9 00 34 PM" src="https://github.com/user-attachments/assets/ae4bca8c-d2fa-4f07-9631-d20173623f8d" />

Reply:
Another idea where the map would still be stored as an array without hashing is to have an empty static EnumMap that you construct the new map referencing. Then it wouldn't have to recalculate the key universe